src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -790,6 +791,75 @@ public UnspentCoinModel[] ListUnspent(int minConfirmations = 1, int maxConfirmat
             return unspentCoins.ToArray();
         }
 
+        [ActionName("listtransactions")]
+        [ActionDescription("Returns an array of transactions belonging to this wallet.")]
+        public TransactionInfoModel[] ListTransactions(string label = "*", int count = 10, int skip = 0)
+        {
+            WalletAccountReference accountReference = this.GetWalletAccountReference();
+
+            var transactions = this.walletManager.GetAllTransactionsInWallet(accountReference.WalletName);
+
+            var filteredTransactions = transactions
+                    .Skip(skip)
+                    .Take(count)
+                    .Select(tx => new TransactionInfoModel
+                    {
+                        BlockHash = tx.BlockHash.ToString(),
+                        Amount = tx.Amount,
+                        TxId = tx.Id.ToString(),
+                        BlockHeight = tx.BlockHeight ?? 0,
+                        Generated = tx.IsCoinBase ?? false,
+                        Confirmations = tx.BlockHeight.HasValue ? (this.walletManager.LastBlockHeight() + 1) - tx.BlockHeight.Value : 0,
+                        Index = tx.Index
```

Comment:
As mentioned, this is actually supposed to be the position of the transaction within the block, not the index of the UTXO being tracked by the wallet. I think `TransactionData` does include a `BlockIndex` property that can be used?